WordPress/Classes/Services/SiteAssemblyService.swift

```diff
@@ -33,21 +33,17 @@ final class EnhancedSiteCreationService: LocalCoreDataService, SiteAssemblyServi
     /// The most recently created blog corresponding to the site creation request; `nil` otherwise.
     private(set) var createdBlog: Blog?
 
-    // MARK: LocalCoreDataService
+    private var coreDataStack: CoreDataStackSwift
 
-    override init(managedObjectContext context: NSManagedObjectContext) {
-        self.accountService = AccountService(coreDataStack: ContextManager.sharedInstance())
-        self.blogService = BlogService(coreDataStack: ContextManager.sharedInstance())
+    init(coreDataStack: CoreDataStackSwift) {
+        self.coreDataStack = coreDataStack
+        self.accountService = AccountService(coreDataStack: coreDataStack)
+        self.blogService = BlogService(coreDataStack: coreDataStack)
 
-        let api: WordPressComRestApi
-        if let wpcomApi = (try? WPAccount.lookupDefaultWordPressComAccount(in: context))?.wordPressComRestApi {
-            api = wpcomApi
-        } else {
-            api = WordPressComRestApi.defaultApi(userAgent: WPUserAgent.wordPress())
-        }
+        let api: WordPressComRestApi = coreDataStack.performQuery { context in
+            (try? WPAccount.lookupDefaultWordPressComAccount(in: context))?.wordPressComRestApi
+        } ?? WordPressComRestApi.defaultApi(userAgent: WPUserAgent.wordPress())
```

Comment:
Out of scope but this made me realize:

- Are we using the same user agent for WordPress and Jetpack? 
- We might want to write a convenience method that does the `WPUserAgent.wordPress()` (or multi-app alternative) for us. E.g. `WordPressComRestApi.defaultApi()`

Reply:
> Are we using the same user agent for WordPress and Jetpack?

I think that is indeed the case. I'm not sure if it's intentional or not...